LIGitHubPR.js

```diff
@@ -0,0 +1,432 @@
+
+    document.getElementsByTagName('body')[0].appendChild(mainPanelElement);
+
+    var fileList = document.getElementsByClassName('pr-file-tree')[0];
+
+    var files = collectFiles();
+
+    for (var i = 0; i < files.length; i++) {
+        displayItem(fileList, files[i], 0);
+    }
+
+    var fileLinks = document.querySelectorAll('.pr-file-tree a');
+
+    for (fileLink of fileLinks) {
+        fileLink.addEventListener('click', function() {
+            this.style.opacity = 0.5;
+        });
+    }
+
+    var refreshHtml = '<a href="#" class="file-tree-refresh">' +
+        '&#128472; Refresh' +
+        '</a>';
+    var refreshElement = createElementFromHTML(refreshHtml);
+
+    mainPanelElement.insertBefore(refreshElement, mainPanelElement.firstChild);
+
+    refreshElement.addEventListener('click', function(e) {
+        e.preventDefault();
+
+        document.querySelectorAll('body')[0].removeChild(document.querySelectorAll('.pr-file-tree')[0]);
+
+        buildUI();
+    });
+}
+
+function displayItem(container, item, level) {
+
+    var fileName = item.path.substring(item.path.length - (item.name.length + 1), item.path.length);
+    var itemHtml = '<div class="item item-' + level + '">';
+
+    if (item.isFile == true) {
+        itemHtml += '<a href="' + item.link + '" title="' + item.path + '">' + '&nbsp;'.repeat(level * 3) + '&nbsp;' + icons.file + '&nbsp;' + item.name + '</a>';
+    } else {
+        itemHtml += '<span style="font-weight:bold;">' + '&nbsp;'.repeat(level * 3) + icons.folder + '&nbsp;' + item.name + '</span>';
+    }
+
+    itemHtml += '</div>';
+    itemCount++;
+
+    var itemElement = createElementFromHTML(itemHtml);
+    if (item.isFile == false && level > 0) {
+        itemElement.classList.add('collapsed');
+        itemElement.classList.add('item-dir');
+        itemElement.querySelectorAll('span')[0].addEventListener('click', function(e) {
+            var elem = e.currentTarget.parentNode;
+            if (hasClass(elem, 'collapsed')) {
+                elem.classList.add('expanded');
+                elem.classList.remove('collapsed');
+            } else {
+                elem.classList.add('collapsed');
+                elem.classList.remove('expanded');
+            }
+        });
+    } else if (item.isFile == false && level == 0) {
+        itemElement.querySelectorAll('span')[0].innerHTML = '';
+    }
+
+    container.appendChild(itemElement);
+
+    sortTreeItems(item.children);
+
+    if (item.children.length > 0) {
+        for (var i = 0; i < item.children.length; i++) {
+            displayItem(itemElement, item.children[i], level + 1);
+        }
+    }
+}
+
+
+function sortTreeItems(items) {
+    items.sort(function(a, b) {
+        let isFileA = a.isFile;
+        let isFileB = b.isFile;
+
+        let nameA = a.name;
+        let nameB = b.name;
+
+        if (isFileA == isFileB) {
+            if (nameA > nameB) return 1;
+            if (nameA < nameB) return -1;
```

Comment:
Ah ah ! String comparison :) Javascript is great ! :) :)

Reply:
You can do the same strange thing with PHP  :scream: